tasks/config_save.rb

```diff
@@ -1,105 +1,23 @@
 #!/opt/puppetlabs/puppet/bin/ruby
-
-require 'json'
 require 'puppet'
-require 'puppet/util/network_device/config'
-require 'timeout'
-
-def require_module_file(file)
-  require "#{Puppet[:plugindest]}/#{file}"
-end
-
-# PARAMETERS: JSON object from STDIN with the following fields:
-#
-# - target: a required string parameter.
-# - noop: an optional boolean metaparameter, will set params['_noop'].
 
-# Read parameters, set defaults, and validate values.
+require_relative '../../ruby_task_helper/files/task_helper.rb'
 
-def read_parameters
-  params = read_stdin
-  return_error("Parameter 'target' contains illegal characters") unless safe_string?(params['target'])
-  params
-end
-
-# Read parameters as JSON from STDIN.
-
-def read_stdin
-  params = {}
-  begin
-    Timeout.timeout(3) do
-      params = JSON.parse(STDIN.read)
+# Bolt task for saving the running config for a Cisco Ios switch
+class ConfigSave < TaskHelper
+  def task(_params, remote, **_kwargs)
+    unless Puppet.settings.global_defaults_initialized?
+      Puppet.initialize_settings
```

Comment:
This doesn't seem to be panos-specific. it might be remote-specific?  -ish? Do we also want to push that to the ruby_task_helper, maybe as optional utility method?

Reply:
If you think it's useful, it would probably be best residing in TaskHelper.